awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.jsx

```diff
@@ -79,39 +88,76 @@ function WorkflowJobTemplateForm({
+          )}
+        >
           <InventoryLookup
-            value={inventoryField.value}
-            isValid={!inventoryMeta.error}
-            helperTextInvalid={inventoryMeta.error}
+            value={inventory}
+            onBlur={() => inventoryHelpers.setTouched()}
             onChange={value => {
-              inventoryHelpers.setValue(value || null);
+              inventoryHelpers.setValue(value.id);
+              setInventory(value);
             }}
+            required
+            touched={inventoryMeta.touched}
+            error={inventoryMeta.error}
           />
-        </FormGroup>
-        <FormField
-          type="text"
-          name="limit"
-          id="wfjt-limit"
+          {(inventoryMeta.touched || formikValues.ask_inventory_on_launch) &&
+            inventoryMeta.error && (
+              <div
+                className="pf-c-form__helper-text pf-m-error"
+                aria-live="polite"
+              >
+                {inventoryMeta.error}
+              </div>
+            )}
+        </FieldWithPrompt>
+
+        <FieldWithPrompt
+          fieldId="wjft-limit"
           label={i18n._(t`Limit`)}
-          tooltip={i18n._(
-            t`Provide a host pattern to further constrain the list of hosts that will be managed or affected by the workflow. This limit is applied to all job template nodes that prompt for a limit. Refer to Ansible documentation for more information and examples on patterns.`
-          )}
-        />
-        <FormField
-          type="text"
+          promptId="template-ask-limit-on-launch"
+          promptName="ask_limit_on_launch"
+          tooltip={i18n._(t`Provide a host pattern to further constrain
+                  the list of hosts that will be managed or affected by the
+                  playbook. Multiple patterns are allowed. Refer to Ansible
+                  documentation for more information and examples on patterns.`)}
+        >
+          <TextInput
+            id="wfjt-limit"
+            {...limitField}
+            isValid={!limitMeta.touched || !limitMeta.error}
+            onChange={(value, event) => {
+              limitField.onChange(event);
+            }}
+          />
+        </FieldWithPrompt>
+
+        <FieldWithPrompt
+          fieldId="wfjt-scm-branch"
           label={i18n._(t`Source Control Branch`)}
+          promptId="wfjt-ask-scm-branch-on-launch"
+          promptName="ask_scm_branch_on_launch"
           tooltip={i18n._(
             t`Select a branch for the workflow. This branch is applied to all job template nodes that prompt for a branch.`
           )}
-          id="wfjt-scm_branch"
-          name="scm_branch"
-        />
+        >
+          <TextInput
+            id="wfjt-scm-branch"
+            {...scmField}
+            onChange={(value, event) => {
+              scmField.onChange(event);
```

Comment:
use setValue here as it imperatively sets the value for the field.  onChange is used for more general purposes.